src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs

```diff
@@ -130,35 +130,14 @@ private Symbol GetSymbolForLocation(Symbol symbol)
             Func<UnassignedFieldsWalker, Symbol, Symbol> getSymbolForLocation,
             DiagnosticBag diagnostics)
         {
-            foreach (var member in members)
+            foreach (var field in members.OfType<FieldSymbol>())
             {
-                if (member.IsStatic != isStatic)
-                {
-                    continue;
-                }
-                TypeWithAnnotations fieldType;
-                FieldSymbol field;
-                switch (member)
-                {
-                    case FieldSymbol f:
-                        fieldType = f.TypeWithAnnotations;
-                        field = f;
-                        break;
-                    case EventSymbol e:
-                        fieldType = e.TypeWithAnnotations;
-                        field = e.AssociatedField;
-                        if (field is null)
-                        {
-                            continue;
-                        }
-                        break;
-                    default:
-                        continue;
-                }
-                if (field.IsConst)
+                if (field.IsStatic != isStatic || field.IsConst)
                 {
                     continue;
                 }
+
+                var fieldType = field.TypeWithAnnotations;
```

Comment:
To be perfectly equivalent to the original code, this would be:

```cs
var fieldType = (field.AssociatedField as EventSymbol)?.TypeWithAnnotations ?? field.TypeWithAnnotations;
```

But it seems unlikely that they would differ.